### ISSUE TITLE
Error in channel dimensions?

### ISSUE THREAD

**Issue by NikolaiKyhne** [MyParadise21/Mamba-SEUNet#8]
Hi MyParadise21,

While reading your paper, you set C2 = C1/2 and C3=C1/3. However, in your code you set C2 = 2*C1 and C3=3*C1.

It happens in line 160 in generator:
self.dim = [cfg['model_cfg']['hid_feature'], cfg['model_cfg']['hid_feature'] * 2, cfg['model_cfg']['hid_feature'] * 3].

Additionally, you add mag and phase refinement blocks. They are not present in the paper:
            nn.Conv2d(dim[0], dim[0], kernel_size=3, stride=1, padding=1, bias=False),

        )

        # 相位
        self.pha_patch_embed_refinement = Patch_Embed_stage(dim[0], dim[0])

        self.pha_refinement = nn.ModuleList([TFMambaBlock(cfg, dim[0]) for _ in range(self.num_tscblocks)])

        self.pha_output = nn.Sequential(
            nn.Conv2d(dim[0], dim[0], kernel_size=3, stride=1, padding=1, bias=False),

        )

        # Initialize decoders
        self.mask_decoder = MagDecoder(cfg)
        self.phase_decoder = PhaseDecoder(cfg)

Can you help me figure out if that's an error in the code or the paper?

Best,

NikolaiKyhne

**Comment by NikolaiKyhne** [comment]
Figured it out myself.
